apps/api/src/app/auth/e2e/login.e2e.ts

```diff
@@ -13,154 +13,179 @@ describe('User login - /auth/login (POST)', async () => {
+        });
+      }
 
-    for (let i = 0; i < MAX_LOGIN_ATTEMPTS; i++) {
-      await session.testAgent.post('/v1/auth/login').send({
+      const { body } = await session.testAgent.post('/v1/auth/login').send({
         email: userCredentials.email,
-        password: 'wrong-password',
+        password: userCredentials.password,
       });
-    }
 
-    const { body } = await session.testAgent.post('/v1/auth/login').send({
-      email: userCredentials.email,
-      password: userCredentials.password,
+      expect(body.statusCode).to.equal(401);
+      expect(body.message).to.contain('Account blocked');
     });
 
-    expect(body.statusCode).to.equal(401);
-    expect(body.message).to.contain('Account blocked');
-  });
-
-  it('should reset the account blocked error after 5 minutes and allow for more 5 failed attempts', async () => {
-    const MAX_LOGIN_ATTEMPTS = 5;
-    const BLOCKED_PERIOD_IN_MINUTES = 5;
+    it('should reset the account blocked error after 5 minutes and allow for more 5 failed attempts', async () => {
+      const MAX_LOGIN_ATTEMPTS = 5;
+      const BLOCKED_PERIOD_IN_MINUTES = 5;
 
-    const lastFailedAttempt = subMinutes(new Date(), BLOCKED_PERIOD_IN_MINUTES);
+      const lastFailedAttempt = subMinutes(new Date(), BLOCKED_PERIOD_IN_MINUTES);
 
-    const failedLogin = {
-      lastFailedAttempt: lastFailedAttempt.toISOString(),
-      times: MAX_LOGIN_ATTEMPTS,
-    };
+      const failedLogin = {
+        lastFailedAttempt: lastFailedAttempt.toISOString(),
+        times: MAX_LOGIN_ATTEMPTS,
+      };
 
-    await userRepository.update(
-      {
-        _id: session.user._id,
-      },
-      {
-        $set: {
-          failedLogin,
+      await userRepository.update(
+        {
+          _id: session.user._id,
         },
+        {
+          $set: {
+            failedLogin,
+          },
+        }
+      );
+
+      for (let i = 0; i < MAX_LOGIN_ATTEMPTS - 1; i++) {
+        const { body } = await session.testAgent.post('/v1/auth/login').send({
+          email: session.user.email,
+          password: 'wrong-password',
+        });
+
+        expect(body.message).to.contain('Incorrect email or password provided.');
+        expect(body.statusCode).to.equal(401);
       }
-    );
 
-    for (let i = 0; i < MAX_LOGIN_ATTEMPTS - 1; i++) {
       const { body } = await session.testAgent.post('/v1/auth/login').send({
-        email: session.user.email,
-        password: 'wrong-password',
+        email: userCredentials.email,
+        password: userCredentials.password,
       });
 
-      expect(body.message).to.contain('Incorrect email or password provided.');
       expect(body.statusCode).to.equal(401);
-    }
+      expect(body.message).to.contain('Account blocked');
+    });
+  });
 
-    const { body } = await session.testAgent.post('/v1/auth/login').send({
-      email: userCredentials.email,
-      password: userCredentials.password,
+  context('with OAuth', async () => {
```

Comment:
All the above ☝️ are just indentation changes after adding a `context`.

Reply:
A new one for me! What is the difference between context and describe?

Reply:
`context` is a synonym for `describe` that is usually available in BDD style APIs. For example https://www.coreycleary.me/better-test-structuring-using-mochas-context